SSA/Core/EDSLNested.lean

```diff
@@ -0,0 +1,257 @@
+import SSA.Core.WellTypedFramework
+/-
+an EDSL that elaborates expressions with nesting
+allowed into SSA.
+
+Authors: Siddharth Bhat
+-/
+namespace EDSL2
+open Lean Elab Macro
+
+declare_syntax_cat dsl_op2
+scoped syntax "[dsl_op2|" dsl_op2 "]" : term
+syntax dsl_var2 := "%v" num
+declare_syntax_cat dsl_expr2
+declare_syntax_cat dsl_region2
+declare_syntax_cat dsl_stmt2
+
+/-
+This is a fancier syntax, which allows expressions in the
+intermediate nodes. This is elaborated into SSA, with temporary
+variable names.
+-/
+syntax dsl_var2 : dsl_expr2
+scoped syntax "()" : dsl_expr2
+scoped syntax "(" dsl_expr2 ")" : dsl_expr2
+scoped syntax "(" dsl_expr2 "," dsl_expr2 ")" : dsl_expr2
+scoped syntax "(" dsl_expr2 "," dsl_expr2 "," dsl_expr2 ")" : dsl_expr2
+scoped syntax "op:" dsl_op2 ws dsl_expr2 (ws dsl_region2)? : dsl_expr2
+syntax dsl_assign2 := dsl_var2 ":= " dsl_expr2
+scoped syntax sepBy(dsl_assign2, ";") : dsl_stmt2
+
+syntax dsl_bb2 := (dsl_stmt2)? "return " dsl_expr2
```

Comment:
Interesting, I didn't know you could declare and define syntax categories in a single go